clients/src/main/java/org/apache/kafka/clients/producer/internals/SenderMetricsRegistry.java

```diff
@@ -0,0 +1,124 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.producer.internals;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.kafka.common.MetricNameTemplate;
+
+public class SenderMetricsRegistry {
+
+    public MetricNameTemplate batchSizeAvg;
+    public MetricNameTemplate batchSizeMax;
+    public MetricNameTemplate compressionRateAvg;
+    public MetricNameTemplate recordQueueTimeAvg;
+    public MetricNameTemplate recordQueueTimeMax;
+    public MetricNameTemplate requestLatencyAvg;
+    public MetricNameTemplate requestLatencyMax;
+    public MetricNameTemplate produceThrottleTimeAvg;
+    public MetricNameTemplate produceThrottleTimeMax;
+    public MetricNameTemplate recordSendRate;
+    public MetricNameTemplate recordsPerRequestAvg;
+    public MetricNameTemplate recordRetryRate;
+    public MetricNameTemplate recordErrorRate;
+    public MetricNameTemplate recordSizeMax;
+    public MetricNameTemplate recordSizeAvg;
+    public MetricNameTemplate requestsInFlight;
+    public MetricNameTemplate metadataAge;
+    public MetricNameTemplate topicRecordSendRate;
+    public MetricNameTemplate topicByteRate;
+    public MetricNameTemplate topicCompressionRate;
+    public MetricNameTemplate topicRecordRetryRate;
+    public MetricNameTemplate topicRecordErrorRate;
+    public MetricNameTemplate batchSplitRate;
+
+    public SenderMetricsRegistry() {
+        this(new HashSet<String>());
+    }
+
+    public SenderMetricsRegistry(Set<String> tags) {
+
+        /* ***** Client level *****/
+        String metricGrpName = "producer-metrics";
+        
+        this.batchSizeAvg = new MetricNameTemplate("batch-size-avg", metricGrpName, "The average number of bytes sent per partition per-request.", tags);
+        this.batchSizeMax = new MetricNameTemplate("batch-size-max", metricGrpName, "The max number of bytes sent per partition per-request.", tags);
+        this.compressionRateAvg = new MetricNameTemplate("compression-rate-avg", metricGrpName, "The average compression rate of record batches.", tags);
+        this.recordQueueTimeAvg = new MetricNameTemplate("record-queue-time-avg", metricGrpName, "The average time in ms record batches spent in the record accumulator.", tags);
```

Comment:
nit: maybe rephrase `record accumulator` to `send buffer` to be more understandable, as record accumulator is an internal classname?